docs/source/_templates/sdk_TP_tutorial.rst

```diff
@@ -195,317 +216,971 @@ Data
+        @staticmethod
+        def from_bytes(payload):
+            return XoPayload(payload=payload)
+
+        @property
+        def name(self):
+            return self._name
+
+        @property
+        def action(self):
+            return self._action
+
+        @property
+        def space(self):
+            return self._space
 
 {% endif %}
 
+Game logic
+==========
 
-Before we say how exactly the transaction payload will be decoded, let's look at
-{% if language == 'JavaScript' %}``_getStateData``{% elif language == 'Java' %}
-``getStateData``{% else %}``_get_state_data``{% endif %}. Now, as far as the handler
-is concerned, it doesn't matter how the game data is stored. The only thing that matters
-is that given a game name, the state store is able to give back the correct game data.
-(In our full XO implementation, the game data is stored in a Merkle-Radix tree.)
+The validation rules and state updates that are associated with 'create', 'delete', and 'take'
+are shown below.
 
 
+Create:
 {% if language == 'JavaScript' %}
 
 .. code-block:: javascript
 
-    const _getStateData = (gameName, stateStore) => {
-        let address = _makeGameAddress(gameName)
+    if (payload.action === 'create') {
+      return xoState.getGame(payload.name)
+        .then((game) => {
+          if (game !== undefined) {
+            throw new InvalidTransaction('Invalid Action: Game already exists.')
+          }
 
-        return stateStore.get([address])
-        .then((stateEntries) => {
-        try {
-          let data =  _decodeData(stateEntries[address])
-          if (data.length < 5){
-            while (data.length < 5){
-              data.push("")
-            }
+          let createdGame = {
+            name: payload.name,
+            board: '---------',
+            state: 'P1-NEXT',
+            player1: '',
+            player2: ''
           }
-          return {board: data[0],
-                  gameState: data[1],
-                  player1: data[2],
-                  player2: data[3],
-                  storedName: data[4]}
-        } catch(err) {
-          throw new InternalError("Failed to deserialize game data." + err)
-        }
-      })
-      .catch(_toInternalError)
+
+          _display(`Player ${player.toString().substring(0, 6)} created game ${payload.name}`)
+
+          return xoState.setGame(payload.name, createdGame)
+        })
     }
 
-    const _toInternalError = (err) => {
-      let message = (err.message) ? err.message : err
-      throw new InternalError(message)
+{% elif language == 'Go' %}
+
+.. code-block:: go
+
+    case "create":
+		err := validateCreate(xoState, payload.Name)
+		if err != nil {
+			return err
+		}
+		game := &xo_state.Game{
+			Board:   "---------",
+			State:   "P1-NEXT",
+			Player1: "",
+			Player2: "",
+			Name:    payload.Name,
+		}
+		displayCreate(payload, player)
+        return xoState.SetGame(payload.Name, game)
+
+With 'validateCreate' defined as
```

Comment:
For all the "With 'X' defined as" in the Go sections, I think you should use double back-ticks. Otherwise the generated HTML looks weird.